gicisky_tag/writer.py

```diff
@@ -14,6 +14,8 @@ def __init__(self, device, image):
         self.image = image
         self.block_size = None
         self.pending_notify = None
+        self.transfer_event = None
+        self.transfer_block = None
```

Comment:
Instead of two fields, could we use a single `asyncio.Queue` containing the blocks to be transferred?  `None` in the channel would signal that the `transfer_task` should return because the transfer is complete.

Reply:
Great idea, I've updated to `asyncio.Queue()`